third_party/libunwindstack/MapInfo.cpp

```diff
@@ -176,6 +192,143 @@ Memory* MapInfo::GetFileMemory() {
   return nullptr;
 }
 
+// Unlike loadable sections of ELF files, the .text section (and other sections) of a Portable
+// Executable can have an offset in the file (PointerToRawData, multiple of FileAlignment) that is
+// not congruent to the offset of that section when loaded into memory (VirtualAddress, multiple of
+// SectionAlignment) modulo the page size. This doesn't fulfill the requirements on the arguments of
+// mmap, so in these cases Wine cannot create a file-backed mapping for the .text section, and
+// resorts to creating an anonymous mapping and copying the .text section into it. This means that,
+// for PE binaries with this property, we cannot simply associate an executable mapping to the
+// corresponding file using the path in the mapping.
+//
+// However, we can make an educated guess. The path of a PE will at least appear in the read-only
+// mapping that corresponds to the beginning of the file, which contains the headers (because the
+// offset in the file is zero and the address chosen for this mapping should always be a multiple of
+// the page size). Therefore, let's consider the file of the last file mapping that precedes this
+// anonymous mapping. If this file is a PE, and an executable file mapping for the .text section of
+// this PE is not present, and finally the offset and the size of this anonymous mapping are
+// compatible with the address range where the .text section would be loaded based on the header for
+// the section (in particular, VirtualAddress and VirtualSize), then we can be quite sure that this
+// anonymous mapping indeed contains the .text section of that PE. Note that we assume that a PE
+// only has one .text section and one executable mapping.
+//
+// This method is the counterpart of GetFileMemory for anonymous executable mappings and contains
+// the detection mechanism described.
+Memory* MapInfo::GetFileMemoryFromAnonExecMapIfPeCoffTextSection() {
+  // We expect this method to be called only for non-special anonymous executable mappings.
+  CHECK(name().empty())
```

Comment:
I realize it's not necessary but in other places CHECK statements end with a semicolon. Add for consistency? (Or is some linter complaining?)

Reply:
Yeah, for consistency I would prefer the ; to be added.